dkg/dkg.go

```diff
@@ -191,10 +201,17 @@ func Run(ctx context.Context, conf Config) (err error) {
 	// Write keystores, deposit data and cluster lock files after exchange of partial signatures in order
 	// to prevent partial data writes in case of peer connection lost
 
-	if err := writeKeystores(conf.DataDir, shares); err != nil {
-		return err
+	if conf.KeymanagerAddr != "" { // Save to keymanager
+		if err = writeKeysToKeymanager(ctx, conf.KeymanagerAddr, shares); err != nil {
```

Comment:
what happens if keys have already been pushed in a previous DKG attempt?

Reply:
A new DKG attempt creates fresh keys so new keys are pushed to keymanager and previously pushed keys would not be overwritten (are not affected).